File: src-tauri/src/commands/settings_commands.rs
use tauri::State;

use crate::application::settings::read_settings_use_case;
use crate::application::settings::update_settings_use_case;
use crate::bootstrap::app_state::AppState;
use crate::domain::settings::app_settings::{
    AppSettings, AppSettingsUpdate, AppSettingsUpdateResult,
};
use crate::shared::error::command_error_mapper::{map_command_error, CommandErrorPayload};

#[tauri::command]
pub fn read_settings(state: State<'_, AppState>) -> Result<AppSettings, CommandErrorPayload> {
    read_settings_use_case::read_settings(state.settings_repository()).map_err(|error| {
        tracing::warn!(?error, "settings read command failed");
        map_command_error(&error)
    })
}

#[tauri::command]
pub fn update_settings(
    state: State<'_, AppState>,
    update: AppSettingsUpdate,
) -> Result<AppSettingsUpdateResult, CommandErrorPayload> {
    update_settings_use_case::update_settings(state.settings_repository(), update).map_err(
        |error| {
            tracing::warn!(?error, "settings update command failed");
            map_command_error(&error)
        },
    )
}

File: src/features/settings/api/settings.commands.ts
import { invokeTauriCommand } from "../../../core/api/tauri-client";
import type {
  AppSettings,
  AppSettingsUpdate,
  AppSettingsUpdateResult,
} from "../types/settings.types";

export function readSettings() {
  return invokeTauriCommand<AppSettings>("read_settings");
}

export function updateSettings(update: AppSettingsUpdate) {
  return invokeTauriCommand<AppSettingsUpdateResult>("update_settings", { update });
}
